Zend/zend_portability.h

```diff
@@ -263,6 +263,18 @@ char *alloca();
 # define ZEND_ATTRIBUTE_UNUSED
 #endif
 
+#if ZEND_GCC_VERSION >= 3000 || __has_attribute(pure)
+# define ZEND_ATTRIBUTE_PURE __attribute__((pure))
+#else
+# define ZEND_ATTRIBUTE_PURE
+#endif
+
+#if ZEND_GCC_VERSION >= 3000 || __has_attribute(const)
+# define ZEND_ATTRIBUTE_CONST __attribute__((const))
```

Comment:
I think you can find it line 232

Reply:
Oh WTF, how could I miss that - I guess I only looked for ``pure`` because that's the one that's actually useful 99% of the time. ``const`` is better but is rarely ever applicable.